pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -1416,6 +1419,15 @@ const (
 	CoreDNSAutoscalingModeClusterProportional CoreDNSAutoscalingMode = "cluster-proportional"
 )
 
+// CoreDNSRewriting contains the setting related to rewriting requests, which are obviously incorrect due to the unnecessary application of the search path.
+type CoreDNSRewriting struct {
+	// Enabled indicates whether rewriting of requests, which are obviously incorrect due to the unnecessary application of the search path, is enabled or not.
+	Enabled bool `json:"enabled" protobuf:"varint,1,opt,name=enabled"`
+	// CommonSuffixes are expected to be the suffix of fully qualified domain name. Each suffix should contain at least one or two dots ('.') to prevent accidental clashes.
+	// +optional
+	CommonSuffixes []string `json:"commonSuffixes,omitempty" protobuf:"bytes,2,rep,name=commonSuffixes"`
```

Comment:
What are the common suffixes used for?

Reply:
(see above)
Depending on the scenario, there may be certain dns suffixes, which might be used regularly. These can be specified in `commonSuffixes` to be excluded from the search path. Assuming that `github.com` is specified in `commonSuffixes`, this would rewrite a request like `www.github.com.my-random-ns.svc.cluster.local` to `www.github.com`.
`commonSuffixes` can be used for internet domains which are commonly used in a way that does not exceed `ndots=5` (number of dots in the name). Without this, the whole search path would be used for the domain resolution.